get_deps.py

```diff
@@ -30,11 +30,24 @@ def main():
 		if dll_name.startswith('???'):
 			print('Unknown DLL?')
 			continue
-		if dll_path.lower().startswith('/c/windows') or dll_path.lower().startswith('c:/windows'):
+		try:
+			# ldd on msys2 gives Unix-style paths, but Python wants them Windows-style
+			# Try cygpath to convert them
+			dll_path = subprocess.check_output(['cygpath', '-w', dll_path]).strip()
+		except:
+			# cygpath doesn't exist or failed. Carry on and hope for the best
+			pass
+		if dll_path.lower().startswith('c:\\windows'):
 			print('Skipping system DLL %s' % dll_path)
 			continue
+
 		print('Copying %s...' % dll_path)
-		shutil.copyfile(dll_path, os.path.join(args.dest, dll_name))
+		try:
+			shutil.copyfile(dll_path, os.path.join(args.dest, dll_name))
+		except shutil.Error as err:
+			# This exception only gets raised if both files are exactly the same, but

```

Comment:
Like above, not a fan of swallowing exceptions, especially when we *know* why this exception will be raised. According to the [documentation](https://docs.python.org/2/library/shutil.html#shutil.copyfile):

> If src and dst are the same files, Error is raised. The destination location must be writable; otherwise, an IOError exception will be raised. If dst already exists, it will be replaced.

I don't read this as "files are exactly the same" (meaning contents), rather the file paths are the same. Is `ldd` returning paths to the libraries copied into the `dist` directory? If yes, then this would explain why the error message is being generated. In this case I would say either (a) we copy a fresh version of the lib from the lib directory or (b) skip copying if the file path already exists.